spring-data-jdbc/src/test/java/org/springframework/data/jdbc/repository/JdbcRepositoryIdGenerationIntegrationTests.java

```diff
@@ -107,15 +131,126 @@ void manuallyGeneratedIdForSaveAll() {
 		assertThat(immutableWithManualIdEntityRepository.findAll()).hasSize(2);
 	}
 
+	@Test // DATAJDBC-2003
+    @Sql(statements = "INSERT INTO EntityWithSeq(id, name) VALUES(1, 'initial value');")
+	void testUpdateAggregateWithSequence() {
+
+        // given.
+        EntityWithSeq entityWithSeq = new EntityWithSeq();
+        entityWithSeq.id = 1L;
+        entityWithSeq.name = "New name";
+
+        // when.
+        EntityWithSeq updated = entityWithSeqRepository.save(entityWithSeq);
+
+        // then.
+        auditableNamedParameterJdbcTemplate.assertSequenceOfStatements(UPDATE);
+        Assertions.assertThat(updated.id).isEqualTo(1L);
+    }
+
+	@Test // DATAJDBC-2003
+	void whenInsertPersistableAggregateWithSequence_thenClientDefinedIdIsFavored() {
+
+        // given.
+        long initialId = 1L;
+        PersistableEntityWithSeq entityWithSeq = PersistableEntityWithSeq.createNew(initialId, "name");
+
+        // when.
+        PersistableEntityWithSeq saved = persistableEntityWithSeqRepository.save(entityWithSeq);
+
+        // then.
+        Assertions.assertThat(saved.getId()).isEqualTo(initialId);
+
+        // We do not expect the SELECT next value from sequence in case we're doing an INSERT with ID provided by the client
+        auditableNamedParameterJdbcTemplate.assertSequenceOfStatements(INSERT);
+    }
+
+	@Test // DATAJDBC-2003
+	void whenInsertAggregateWithSequenceAndPrimitiveIdUnset_thenSequenceIsQueried() {
+
+        // given.
+        PrimitiveIdEntityWithSeq entity = new PrimitiveIdEntityWithSeq();
+        entity.name = "some name";
+
+        // when.
+        PrimitiveIdEntityWithSeq saved = primitiveIdEntityWithSeqRepository.save(entity);
+
+        // then.
+        Assertions.assertThat(saved.id).isEqualTo(1L); // sequence starts with 1
+
+        // 1. Select from sequence
+        // 2. Actual INSERT
+        auditableNamedParameterJdbcTemplate.assertSequenceOfStatements(SELECT, INSERT);
+    }
+
 	private interface PrimitiveIdEntityRepository extends ListCrudRepository<PrimitiveIdEntity, Long> {}
 
 	private interface ReadOnlyIdEntityRepository extends ListCrudRepository<ReadOnlyIdEntity, Long> {}
 
 	private interface ImmutableWithManualIdEntityRepository extends ListCrudRepository<ImmutableWithManualIdEntity, Long> {}
 
+    private interface EntityWithSeqRepository extends ListCrudRepository<EntityWithSeq, Long> {}
+    private interface PersistableEntityWithSeqRepository extends ListCrudRepository<PersistableEntityWithSeq, Long> {}
+    private interface PrimitiveIdEntityWithSeqRepository extends ListCrudRepository<PrimitiveIdEntityWithSeq, Long> {}
+
 	record ReadOnlyIdEntity(@Id Long id, String name) {
 	}
 
+    static class EntityWithSeq {
```

Comment:
Could be just `WithSeq`

Reply:
Renamed to `SimpleSeq` to align with other entities names. Current entity names are:

- `SimpleSeq`
- `PersistableSeq`
- `PrimitiveIdSeq`